### ISSUE TITLE
Make global.json "sdk/version" support min/max, or wildcards

### ISSUE THREAD

**Issue by kierenj** [dotnet/core-setup#679]
Today, I tried to deploy to Azure App Services, but they had updated to the latest tools SDK (preview3).  This broke the build for my project.

Eventually, I found that I can specify an sdk/version to use a different version of the SDK.  However, App Services have a range of versions installed, and the previous version I had (`1.0.0-preview2-003133`) wasn't an option - they have `1.0.0-preview2-003131`.

I had to add a custom deployment script to copy in a different production version of global.json.

It would be great if there was a version range option of some sort here.  I'd love to specify `1.0.0-preview2-*`.

Moved from https://github.com/dotnet/cli/issues/4794

**Comment by gkhanna79** [comment]
@ramarag @schellap PTAL

**Comment by natemcmaster** [comment]
+1. Would also be nice to have an option such as "preferPrerelease" to opt-in to latest unstable installation even if a stable SDK version is installed. cref https://github.com/dotnet/core-setup/issues/1664#issuecomment-285140280

**Comment by gkhanna79** [comment]
CC @livarcocc as this is in the CLI/SDK policy resolution.

**Comment by dasMulli** [comment]
I suppose this is - or a different roll-forward behavior- is now more important with VS 15.3+ picking up the used SDK by calling into hostfxr through its SDK resolver.

This would mean that installing a daily build of a 2.1.0-** CLI would make a released version of vs 2017 15.3 pick up that version and build all your production projects with a preview SDK without users noticing.

**Comment by onovotny** [comment]
Apparently there is a way to specify "latest prerelease"

```json
{
  "sdk": {
    "version": "3.1.100-preview"
  }
}
```
Because there is no exact version that matches, it'll look for the latest, which would be something like `3.1.100-preview.9754`.

**Comment by dasMulli** [comment]
Also see @vitek-karas' suggestion for NuGet-style version specifications at 
https://github.com/dotnet/core-setup/issues/3623#issuecomment-441772412

**Comment by mmillican** [comment]
To reiterate what @NickCraver said, supporting ranges or at least _minimum_ versions in `global.json` would be huge. Currently, we have ~15-20 ASP.NET Core projects and its extremely hard to keep them all in sync and making sure all devs (and build servers) have the correct version(s) of the SDK installed.

**Comment by mmisztal1980** [comment]
Is there any progress in this area? I've just been hit by this, trying to develop a solution on VS 2019 and VS 2019 on MacOS, 

On MacOS the latest dotnet SDK version I can use is `2.2.107`, which appears to be unavailable for Windows downloads. This is plain and outright stupid.

**Comment by NickCraver** [comment]
Unsure about progress here, but to unblock you can find 2.2.107 for Windows on the full download list: https://dotnet.microsoft.com/download/dotnet-core/2.2

**Comment by mmisztal1980** [comment]
Thanks @NickCraver  no idea why I haven't spotted it earlier

**Comment by mmisztal1980** [comment]
@NickCraver any idea how to install 2.2.107 on a ubuntu box? I followed the instructions but I ended up with a higher versions

**Comment by cmelgarejo** [comment]
This is needed for cross-platform development, Linux has `2.2.300` and MacOS `2.2.107`, our `global.json` file is set to the `.300` SDK and vscode c# plugin on MacOS does not parse the projects correctly then you have to change to `.107`, screwing up everytime you want to develop on MacOS, for the IDE to show no errors, or else there are like 11k + for example 😢

**Comment by dagood** [comment]
You *can* use the binary (`tar.gz`) installer for Linux to get 2.2.107. It's the `.NET Core Binaries` line, e.g. https://dotnet.microsoft.com/download/thank-you/dotnet-sdk-2.2.107-linux-x64-binaries. Not preferred, but the bits at least are there.

You can use `apt list -a dotnet-sdk-2.2` to show the versions available on your distro, or browse:
https://packages.microsoft.com/ubuntu/18.04/prod/pool/main/d/dotnet-sdk-2.2/
https://packages.microsoft.com/ubuntu/19.04/prod/pool/main/d/dotnet-sdk-2.2/
(...and more.)

I'm not able to find a `2.2.107` SDK available as a Deb package. 😕 I opened https://github.com/dotnet/core/issues/2825 to track/discuss this.

Related discussion: https://github.com/dotnet/core/issues/2721 (Installing old SDK versions on CentOS/Fedora)

**Comment by vitek-karas** [comment]
Proposed design which should include this feature request is discussed here: https://github.com/dotnet/designs/pull/71

There's also ongoing implementation in https://github.com/dotnet/core-setup/pull/6953

**Comment by HEskandari** [comment]
This broke a local build of ours when I updated my Visual Studio from VS 16.2.5. to 16.3.4. The reason is that the new update included .NET Core v3 which is what was picked to build our solution. Fixing the .NET Core SDK using a global.json file to say `2.2.402` broke the build on the build agent since that's not the exact SDK version installed there.
